Loosen supported ruby version to 2.7

okuramasafumi opened the pull request.

okuramasafumi committed:
We don't want to force other devs to use specified Ruby

Removing `.ruby-version` file lets other devs to use their preferred
version of Ruby.

okuramasafumi committed:
Loosen Ruby version requirement to 2.7.0

2.7 series is still supported by the Core Team so we can also support it.

IkumaTadokoro commented:
The reason why the CI is failing is that it depends on the `.ruby-version` that we removed.
I will fix it to specify the Ruby execution environment in the workflow(yml), so please wait.

IkumaTadokoro commented:
@okuramasafumi 
I specify the Ruby version in the workflow configuration file instead of `.ruby-version` in #79 .
Could you please rebase main and add `2.7` to the following workflow?

- `.github/workflows`
    - `lint.yml`
    - `test.yml`
    - `steep.yml`

okuramasafumi committed:
Add `2.7` to some GitHub Actions

okuramasafumi committed:
Fix RuboCop warnings

Some strings are already frozen and a range should be frozen.
Command: `bundle exec rubocop -A`

okuramasafumi committed:
Use `JSON.parse` instead of `JSON.load_file`

Though introduced recently in 67bbacfb5ac096110b5bde431e60a659635f4f76 ,
`JSON.load_file` was added in Ruby 3.0 and doesn't work in 2.7.
We need to use `JSON.parse` since we support Ruby 2.7 now.

IkumaTadokoro reviewed (approved):
LGTM!! Thanks for your PR!😊

---

I'll post the link about `JSON.load_file` (ruby 3.0~) again as a reminder.
→ https://docs.ruby-lang.org/en/3.0/JSON.html#method-i-load_file

Note: This method is not described in `rurema`, so I might want to check it separately.
